cmd/pro/add/cluster.go

```diff
@@ -0,0 +1,312 @@
+package add
+
+import (
+	"cmp"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/log/survey"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	managementv1 "github.com/loft-sh/api/v4/pkg/apis/management/v1"
+	storagev1 "github.com/loft-sh/api/v4/pkg/apis/storage/v1"
+	proflags "github.com/loft-sh/devpod/cmd/pro/flags"
+	"github.com/loft-sh/devpod/pkg/config"
+	"github.com/loft-sh/devpod/pkg/loft"
+	"github.com/loft-sh/devpod/pkg/loft/client"
+	"github.com/loft-sh/devpod/pkg/pro"
+	"github.com/loft-sh/devpod/pkg/workspace"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type ClusterCmd struct {
+	Log log.Logger
+	*proflags.GlobalFlags
+
+	Namespace        string
+	ServiceAccount   string
+	DisplayName      string
+	KubeContext      string
+	Insecure         bool
+	Wait             bool
+	HelmChartPath    string
+	HelmChartVersion string
+	HelmSet          []string
+	HelmValues       []string
+	Host             string
+}
+
+// NewClusterCmd creates a new command
+func NewClusterCmd(globalFlags *proflags.GlobalFlags) *cobra.Command {
+	cmd := &ClusterCmd{
+		GlobalFlags: globalFlags,
+		Log:         log.GetInstance(),
+	}
+
+	c := &cobra.Command{
+		Use:   "cluster",
+		Short: "add current cluster to DevPod Pro",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cobraCmd *cobra.Command, args []string) error {
+			devPodConfig, err := config.LoadConfig(cmd.Context, "")
+			if err != nil {
+				return err
+			}
+
+			if cmd.Host == "" {
```

Comment:
what do you think about extracting this to separate function? So that `NewClusterCmd` just creates a new cobra command and doesn't contain actual logic?

Reply:
fair enough, done 👍